packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -39,15 +44,44 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     style = variant === 'accent' || variant === 'cta' ? 'fill' : 'outline',
     staticColor,
     isDisabled,
+    isPending,
     autoFocus,
     ...otherProps
   } = props;
   let domRef = useFocusableRef(ref);
   let {buttonProps, isPressed} = useButton(props, domRef);
   let {hoverProps, isHovered} = useHover({isDisabled});
+  let stringFormatter = useLocalizedStringFormatter(intlMessages);
   let {styleProps} = useStyleProps(otherProps);
+  let progressCircleProps: SpectrumProgressCircleProps = {
+    'aria-label': stringFormatter.format('loading'),
+    isIndeterminate: true,
+    size: 'S',
+    ...(
+      (variant === 'overBackground' && style !== 'fill') ||
+      ((variant === 'cta' || variant === 'accent') && style !== 'outline')
+    ) && {variant: 'overBackground'}
```

Comment:
We will need to double check with design, but it looks like the pending state should also cause the button to become disabled, i.e. gray background. So I guess the spinner would always be blue? https://spectrum.adobe.com/page/button/#Pending

Reply:
I based my changes on the Spectrum Site and an XD file that @dannify sent me. In the XD file the buttons are not in a visually disabled state.

<img width="531" alt="image" src="https://github.com/adobe/react-spectrum/assets/3374077/ef97259c-e001-4dad-a86a-bbf056397e55">